File: Homework3/Question2/src/Chicken.java
public class Chicken extends Bird {
    public Chicken(String color, int numberOfFeathers) {
        super(color, numberOfFeathers);
    }

    public Chicken(String color) {
        this(color, 8000);
    }

    public static void testChicken() {
        Chicken chick1 = new Chicken("yellow", 2160);
        Chicken chick2 = new Chicken("red");
        //Test chick1
        System.out.println(chick1.getColor() == "yellow");
        System.out.println(chick1.getNumberOfFeathers() == 2160);
        System.out.println(chick1.isEdible() == true);
        //Test chick2
        System.out.println(chick2.getColor() == "red");
        System.out.println(chick2.getNumberOfFeathers() == 8000);
        System.out.println(chick2.isEdible() == true);
    }
}

File: Homework3/Chicken.kt
 open class Chicken : Bird {
     constructor(color: String, numberOfFeathers: Int) : super(color, numberOfFeathers)

     constructor(color: String) : super(color, 8000)

     companion object {
         fun testChicken() {
             val chick1 = Chicken("yellow", 2160)
             val chick2 = Chicken("red")
             println(chick1.getColor() == "yellow")
             println(chick1.getNumberOfFeathers() == 2160)
             println(chick1.isEdible() == true)
             println(chick2.getColor() == "red")
             println(chick2.getNumberOfFeathers() == 8000)
             println(chick2.isEdible() == true)
         }
     }
 }